### ISSUE TITLE
Documentation and APIs

### ISSUE THREAD

**Issue by crudbug** [orientechnologies/orientdb#5037]
I am creating a prototype using OrientDB. 

I have realized that the documentation is very inconsistent for starting development for new comers. For example, when discussing Document DB, it suddenly jumps to Graph DB concepts. Same for Object DB concepts. Also, some APIs are deprecated in the examples.

Explicitly specifying different storage models -  Graph DB, Object DB & Document DB and using simple data model with CRUD operations to describe different operations will be a good addition to the documentation. 

Also, there are inconsistencies in APIs.

My mental model thinks when acquiring a txn,  I would acquire either one of 

* OGraphDatabaseTx
* OObjectDatabaseTx
* ODocumentDatabaseTx not ODatabaseDocumentTx
 
When creating an instance from a pool, **OGraphDatabasePool** etc.

**OObjectDatabasePool** is deprecated in 2.1.2, but the replacement **OPartitionedDatabasedPool** does not return a **OObjectDatabaseTx**

Happy to discuss other thoughts.

**Comment by Jotschi** [comment]
I noticed the following points:

* __Multithreading__

Examples on how to use orientdb Java API in a multithreading environment would helped me a lot. It was unclear to me that i can't reuse ```OrientVertex``` instances across multiple threads.
I guess it would not hurt to add javadoc within OrientVertex that informs you that the OrientVertex class is not theadsafe.
 
* __Warnings__

I'm currently creating ```OrientEdgeType's``` on the fly and thus i see a lot of warnings in my log. I did not realize that i should create the ```OrientEdgeType's``` when i initialize my graph or in a dedicated notx transaction. Instead i stumbled upon ```OrientBaseGraph.executeOutsideTx()``` which has no JavaDoc & no documentation. I think it would be good to adapt the warning message or to create a dedicated documentation page which explains the warning and shows ways to avoid those warnings.

```
WARNING: {db=tinkerpop} Requested command 'create edge type 'HAS_ITEM' as subclass of 'E'' must be executed outside active transaction: the transaction will be committed and reopen right after it. To avoid this behavior execute it outside a transaction
```

* __embedding / remote connection__

The http://orientdb.com/docs/2.1/Embedded-Server.html documentation explains how to setup the OServer but it does not explain how to use the OServer in combination with a plocal database in the same JVM. After some research and experimentation i realized that the OServer will automatically add found databases to the server when it detects it in the database folder and that the default login is admin/admin. A section that explains this would be very nice. Additionally a section that also explains how to enable replication in this kind of environment would be nice.

* __Indices__

Examples on how to setup and use Verticle and Edge Indices via Java would be very helpful. I was able to setup the edge indices when i took a look at the performance test mentioned in #4105.
I did not manage to create an index using the ```IndexableGraph.createIndex()``` method. Instead i used the ```OrientEdgeType.createIndex()``` method. In general it is still a bit confusing to me that there are multiple ways to create an index. Furthermore some methods seem to utilize the index and some are not. For example  ```graph.getVertices("item.name", "somename"``` is not working for me but ```graph.getVertices("item", new String[] { "name" }, new Object[] { "somename" })```is.

Initially i got the impression that orientdb would automatically create edge indices and make use of them. For example when invoking: ```OrientVertex.getEdges(final OrientVertex iDestination, final Direction iDirection, final String... iLabels)``` but this is afaik not the case.

* __Edge Schema/Type__

I'm creating verticle types using ```graph.createVertexType("item", "V");``` but i needed to use the label of my edges to create the edge type ```graph.createEdgeType("HAS_ITEM");``` 

* __Documentation versions__

https://www.google.com/search?q=oriendb+index+java returns various versions of documentation pages. I think it would be good to put a info text at the top of each 2.0 and older documentation pages which tells you that there is a newer version accessible.

**Comment by thejeed** [comment]
I'd like to add something as I just wanted to create an issue but then I saw that there's already this one.

* Navigation
As a Chrome user, the documentation is very painful to use as a click at the tree of topics on the left makes this tree disappear. Using the browser "back" sometimes doesn't get me back where I came from, I've got to reload the whole page. Also, the search function is broken. It doesn't return the filtered tree nor does it return any result. The workaround is to use Firefox, but I just don't want to install it just to use it with a single page(...any longer).

* Doc content
Looking at the content, I'd say that there could be a bit more stuff about handling stuff via the Java API. Adding some more best practises could reduce redundant questions in the google group or on Stackoverflow and lighten up the customer experience.

My personal favourite "db-docs" are the ones of PostgreSQL. They've got a nice, clean, working interface that supports several versions of the product, as well:

http://www.postgresql.org/docs/9.3/interactive/rules-views.html#RULES-SELECT

Nevertheless, they don't have such a tree-of-topics representation at each site which I like about the OrientDB docs.
